libp2p/switch.nim

```diff
@@ -365,10 +368,11 @@ proc start*(s: Switch) {.public, async: (raises: [CancelledError, LPError]).} =
       s.acceptFuts.add(s.accept(t))
       s.peerInfo.listenAddrs &= t.addrs
 
+  await s.peerInfo.update()
```

Comment:
I'm not sure about this change. 
I think the reason why it's done this way is because autorelay service will modify the address mappers in  libp2p/services/autorelayservice.nim in line 80
```
    switch.peerInfo.addressMappers.add(self.addressMapper)
``` 

In any case. I guess it wouldnt hurt to call s.peerInfo.update() twice (where you propose to do it, and where it was done before).